src/components/alert/bl-alert.ts

```diff
@@ -0,0 +1,139 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { event, EventDispatcher } from '../../utilities/event';
+import style from './bl-alert.css';
+import '../icon/bl-icon';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { iconConverter } from './icon.converter';
+import { ButtonVariant, ButtonKind, ButtonSize } from '../button/bl-button';
+import { classMap } from 'lit/directives/class-map.js';
+
+export type AlertVariant = 'info' | 'warning' | 'success' | 'error';
+
+/**
+ * @tag bl-alert
+ * @summary Baklava Alert component
+ *
+ * @cssproperty --bl-alert-display - Sets the display property of button. Default value is 'inline-block'.
+ */
+
+@customElement('bl-alert')
+export default class BlAlert extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  @property({ reflect: true })
+  variant: AlertVariant = 'info';
+
+  @property()
+  description: 'string';
+
+  @property({ converter: iconConverter() })
+  icon?: boolean | string;
+
+  @property({ type: Boolean })
+  closable = false;
+
+  @property()
+  caption?: string;
+
+  @state()
+  _hidden = false;
+
+  @event('bl-close') private onClose: EventDispatcher<boolean>;
+
+  get _hasAlertCaptionSlot() {
+    return this.querySelector(':scope > [slot="caption"]') !== null;
+  }
+
+  get _getAlertActionSlot() {
+    return this.querySelector(':scope > [slot="action"]');
+  }
+
+  _closeHandler() {
+    this._hidden = true;
+    this.onClose(true);
+  }
+
+  shouldRender(value: undefined | boolean | string, html: TemplateResult) {
+    if (!value) return null;
+    return html;
+  }
+
+  predefinedIcons() {
+    switch (this.variant) {
+      case 'success':
+        return 'check_fill';
+      case 'error':
+        return 'close_fill';
+      default:
+        return this.variant;
+    }
+  }
+
+  getIcon(): string | undefined {
+    if (!this.icon) return;
+    if (typeof this.icon === 'boolean') {
+      return this.predefinedIcons();
+    }
+    return this.icon;
+  }
+
+  initAlertActionSlot() {
+    const actionSlot = this._getAlertActionSlot;
+    if (!actionSlot) return;
+    if (actionSlot?.tagName !== 'BL-BUTTON')
+      throw new Error('Action slot must contain bl-button component as child!');
```

Comment:
@muratcorlu @leventozen 
While im searching an issue right after refactoring `initAlertActionSlot()` to listen `slotchange` event. 
Found exactly same issue that i've facing [here](https://github.com/open-wc/open-wc/issues/2124) and that thread mentions that throwing an error inside a web component is actually anti-pattern and displaying a message inside component could be a solution. 
What would you suggest for this case?

Reply:
OK, that looks ok for now. If removing action slot creates a big annoyance with some valid arguments, we can consider to allow other type of elements in the future.